pkg/common/fflag/fflag.go

```diff
@@ -81,23 +71,38 @@ func IsSet(f Flag) bool {
 	return singleton.flags[f]
 }
 
-func load(flagConfig map[string]bool) {
+func load(flagConfig map[string]bool, supplementalFlags map[Flag]bool) {
 	singleton.mtx.Lock()
 	defer singleton.mtx.Unlock()
 
+	for f, v := range supplementalFlags {
+		// Common flags should take precedent
```

Comment:
Should they?

Reply:
Hmm, I think taking common is probably not correct? I see us in one of two situations for shared feature flags:

1. Flag defaults must be consistent across server and agent. In this case, I would think it should be an error for the supplemental flag set to overlap with the common flag set. We should just have one flag defined.
2. Flag defaults can be different across server and agent. In this case, if we took common, there would be no way to override the default from the component. I think we'd want to take the supplemental default in this case to allow the server/agent to override the common default.

Reply:
Ah gotcha. Yeah, I think an error makes the most sense :heart: